Modules/Tokens/src/Tokens.Domain/Entities/Token.cs

```diff
@@ -38,24 +39,23 @@ public Token(IdentityAddress createdBy, DeviceId createdByDevice, byte[] content
     public IdentityAddress CreatedBy { get; set; }
     public DeviceId CreatedByDevice { get; set; }
 
+    public IdentityAddress? ForIdentity { get; set; }
+
     public byte[] Content { get; private set; }
     public DateTime CreatedAt { get; set; }
     public DateTime ExpiresAt { get; set; }
 
-    public void LoadContent(byte[] content)
-    {
-        if (Content != null)
-            throw new Exception("Cannot change the content of a token.");
-
-        Content = content;
-    }
-
     public static Expression<Func<Token, bool>> IsExpired =>
         challenge => challenge.ExpiresAt <= SystemTime.UtcNow;
 
     public static Expression<Func<Token, bool>> IsNotExpired =>
         challenge => challenge.ExpiresAt > SystemTime.UtcNow;
 
+    public static Expression<Func<Token, bool>> CanBeCollectedBy(IdentityAddress? address)
+    {
+        return token => token.ForIdentity == null || token.ForIdentity == address || token.CreatedBy == address;
```

Comment:
The way this expression is set up now, when someone attempts to List Tokens, they will be able to see every single token as long as the `ForIdentity` property is not set.
I would imagine this is not the intended, as Tokens should only be collected either by the `Owner` or someone set as `ForIdentity`.
If this is the case, please fix the Expression and also consider adding some integration tests that check for something other than the happy path (e.g., can't retrieve `Token` when neither `Owner` of `ForIdentity`)

Reply:
Actually, the idea is that anyone can pick up a token unless that token has an optional `ForIdentiy` field set. This is also explicitly conformed in the token tests.

Reply:
Are you sure about this?
Before this Pull Request, we would specifically have a condition for the `Owner` (see below):
```
private async Task<DbPaginationResult<Token>> Find(IdentityAddress? owner, IEnumerable<TokenId> ids, PaginationFilter paginationFilter, CancellationToken cancellationToken)
{
    if (paginationFilter == null)
        throw new Exception("A pagination filter has to be provided.");

    var query = _readonlyTokensDbSet.Where(Token.IsNotExpired);

    var idsArray = ids as TokenId[] ?? ids.ToArray();

    if (idsArray.Any())
        query = query.Where(t => idsArray.Contains(t.Id));

    if (owner != null)
        query = query.Where(t => t.CreatedBy == owner);

    var dbPaginationResult = await query.OrderAndPaginate(d => d.CreatedAt, paginationFilter, cancellationToken);

    return dbPaginationResult;
}
```

If this is now not the case, the methods for `FindAllOfOwner` make no sense.
It would instead just be a `FindAll` method that has an optional parameter for the list of Token Ids.

Reply:
I'm fairly certain of this. 
`FindAllOfOwner` might still make sense as you might want to filter your own tokens.

Reply:
But that's my point, `FindAllOfOwner` doesn't filter your own tokens.
With the current expression, if `ForIdentity = null` then you get the token regardless if you're the `Owner` or not.

Reply:
Damn, good catch. This should fix it.